staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -118,11 +118,35 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if err != nil {
 				return true, nil, err
 			}
-			err = tracker.Update(gvr, action.GetObject(), ns)
+
+			var obj runtime.Object
+			switch action.GetSubresource() {
+			case "status":
```

Comment:
Unfortunately, not all resources uniformly prevent writes to everything outside `Status` via the status subresource. Several resources allow modifying metadata (labels, annotations, finalizers, etc)